lib/jxl/encode.cc

```diff
@@ -401,6 +409,12 @@ JxlEncoderStatus JxlEncoderAddImageFrame(const JxlEncoderOptions* options,
     c_current = options->enc->metadata.m.color_encoding;
   }
 
+  // treat multiple frames as animation
+  // until animation API is established
+  options->enc->metadata.m.have_animation = true;
+  options->enc->metadata.m.animation.tps_numerator = 100;
```

Comment:
The encoder API has the function JxlEncoderSetBasicInfo, with which the user can specify the fields have_animation, tps_numerator, tps_denominator, etc... Rather than set it here, a better place would be to set these metadata fields in JxlEncoderSetBasicInfo (also the tps_denominator, num_loops and have_timecodes fields from JxlAnimationHeader)

It looks like now it would always set have_animation, even for a still image. With JxlEncoderSetBasicInfo, the user can already choose whether have_animation is enabled or not, so it should already be possible to choose between a still made up of multiple frames, or an actual animation, that way.

The frame header for frame.duration is not implemented yet in the encoder, so the duration = 10 as a temporary solution is ok, much better would of course be if the user can set a JxlFrameHeader per frame.

Reply:
@lvandeve  I missed that JxlBasicInfo includes animation.  So here should set duration in response to whether animation is already set or not.  If animation is not set, what should happen when a second frame is sent to the encoder?  Should it be rejected (set first frame as the last frame)?

Reply:
When animation is not set, the duration of frames is not signaled, and it is implicitly zero. This is what you would do in case of a multi-layer still image.

Reply:
@jonsneyers The decoders don't work when animation is not set.  The API decodes only one frame, and `djxl` gives an error.

Reply:
The API doesn't provide a way (at least not atm) to extract non-coalesced frames, like the separate layers of a composite still image, it will only return the blended/merged image (which is good for viewing apps that wouldn't know how to properly blend the layers themselves).
But if djxl gives an error then I suppose the encode api somehow produces an invalid bitstream in this case.

Reply:
I think what's missing is to set duration != 0 in the encoder. That could easily be a setting when adding the frame.